### ISSUE TITLE
Cookbook apt not found

### ISSUE THREAD

**Issue by daniel-dgi** [Islandora-Labs/islandora_chef#7]
Updated code to commit 4c570e6d73ad9afb7c499e5eb55c6e5272665f01

Running on mac.  New ruby with rbenv.  Newest vagrant.  Installed both plugins successfully.  Blew away both lock files in repo and am getting this when trying to spin up a box:

daniels-mbp:islandora_chef danny$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/precise64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/precise64' is up to date...
==> default: Setting the name of the VM: islandora_chef_default_1422891069892_80991
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 8080 => 8080 (adapter 1)
    default: 3306 => 3306 (adapter 1)
    default: 5432 => 5432 (adapter 1)
    default: 80 => 8181 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if its present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 4.1.12
    default: VirtualBox Version: 4.3
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => /Users/danny/Git/islandora_chef
    default: /tmp/vagrant-chef/0add50be72e723dbdbbf8210bf1bef58/cookbooks => /Users/danny/Git/islandora_chef/cookbooks
==> default: Installing Chef 11.6.2 Omnibus package...
==> default: Downloading Chef 11.6.2 for ubuntu...
==> default: downloading https://www.chef.io/chef/metadata?v=11.6.2&prerelease=false&nightlies=false&p=ubuntu&pv=12.04&m=x86_64
==> default:   to file /tmp/install.sh.1786/metadata.txt
==> default: trying wget...
==> default: url	https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/12.04/x86_64/chef_11.6.2-1.ubuntu.12.04_amd64.deb
==> default: md5	a809b8cfeaea71c20c0de0e492c7f86f
==> default: sha256	74d5283b597299efbff3c66897c79876cc11aade6dc8810ddf38101526203651
==> default: downloaded metadata file looks valid...
==> default: downloading https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/12.04/x86_64/chef_11.6.2-1.ubuntu.12.04_amd64.deb
==> default:   to file /tmp/install.sh.1786/chef_11.6.2-1.ubuntu.12.04_amd64.deb
==> default: trying wget...
==> default: Comparing checksum with sha256sum...
==> default: Installing Chef 11.6.2
==> default: installing with dpkg...
==> default: Selecting previously unselected package chef.
==> default: (Reading database ...
==> default: 66209 files and directories currently installed.)
==> default: Unpacking chef (from .../chef_11.6.2-1.ubuntu.12.04_amd64.deb) ...
==> default: Setting up chef (11.6.2-1.ubuntu.12.04) ...
==> default: Thank you for installing Chef!
==> default: Running provisioner: chef_solo...
==> default: Detected Chef (latest) is already installed
Generating chef JSON and uploading...
==> default: Running chef-solo...
==> default: stdin: is not a tty
==> default: [2015-02-02T15:31:51+00:00] INFO: Forking chef instance to converge...
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Fork successful. Waiting for new chef pid: 1963
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Forked instance now converging
==> default: Starting Chef Client, version 11.6.2
==> default: [2015-02-02T15:31:51+00:00] INFO: *** Chef 11.6.2 ***
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Building node object for islandora-fedora4
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Extracting run list from JSON attributes provided on command line
==> default: [2015-02-02T15:31:51+00:00] INFO: Setting the run_list to ["recipe[ubuntu-baseline]", "recipe[islandora]"] from JSON
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Applying attributes from json file
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Platform is ubuntu version 12.04
==> default: [2015-02-02T15:31:51+00:00] INFO: Run List is [recipe[ubuntu-baseline], recipe[islandora]]
==> default: [2015-02-02T15:31:51+00:00] INFO: Run List expands to [ubuntu-baseline, islandora]
==> default: [2015-02-02T15:31:51+00:00] INFO: Starting Chef Run for islandora-fedora4
==> default: [2015-02-02T15:31:51+00:00] INFO: Running start handlers
==> default: [2015-02-02T15:31:51+00:00] INFO: Start handlers complete.
==> default: [2015-02-02T15:31:51+00:00] DEBUG: No chefignore file found at /tmp/vagrant-chef/0add50be72e723dbdbbf8210bf1bef58/cookbooks/chefignore no files will be ignored
==> default: Compiling Cookbooks...
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Re-raising exception: Chef::Exceptions::CookbookNotFound - Cookbook apt not found. If you're loading apt from another cookbook, make sure you configure the dependency in your metadata
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/cookbook/cookbook_collection.rb:38:in `block in initialize'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `yield'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `default'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `default'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:265:in `[]'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:265:in `each_cookbook_dep'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:243:in `add_cookbook_with_deps'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:244:in `block in add_cookbook_with_deps'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:266:in `each'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:266:in `each_cookbook_dep'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:243:in `add_cookbook_with_deps'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:87:in `block in cookbook_order'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:85:in `each'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:85:in `cookbook_order'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:97:in `compile_libraries'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:70:in `compile'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context.rb:86:in `load'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:249:in `setup_run_context'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:492:in `do_run'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:199:in `block in run'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:193:in `fork'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:193:in `run'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application.rb:183:in `run_chef_client'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:239:in `block in run_application'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:231:in `loop'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:231:in `run_application'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application.rb:66:in `run'
==> default:   /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/bin/chef-solo:25:in `<top (required)>'
==> default:   /usr/bin/chef-solo:23:in `load'
==> default:   /usr/bin/chef-solo:23:in `<main>'
==> default: [2015-02-02T15:31:51+00:00] ERROR: Running exception handlers
==> default: [2015-02-02T15:31:51+00:00] ERROR: Exception handlers complete
==> default: [2015-02-02T15:31:51+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2015-02-02T15:31:51+00:00] DEBUG: Chef::Exceptions::CookbookNotFound: Cookbook apt not found. If you're loading apt from another cookbook, make sure you configure the dependency in your metadata
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/cookbook/cookbook_collection.rb:38:in `block in initialize'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `yield'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `default'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/ohai-6.18.0/lib/ohai/mash.rb:77:in `default'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:265:in `[]'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:265:in `each_cookbook_dep'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:243:in `add_cookbook_with_deps'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:244:in `block in add_cookbook_with_deps'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:266:in `each'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:266:in `each_cookbook_dep'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:243:in `add_cookbook_with_deps'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:87:in `block in cookbook_order'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:85:in `each'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:85:in `cookbook_order'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:97:in `compile_libraries'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context/cookbook_compiler.rb:70:in `compile'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/run_context.rb:86:in `load'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:249:in `setup_run_context'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:492:in `do_run'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:199:in `block in run'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:193:in `fork'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/client.rb:193:in `run'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application.rb:183:in `run_chef_client'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:239:in `block in run_application'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:231:in `loop'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application/solo.rb:231:in `run_application'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/lib/chef/application.rb:66:in `run'
==> default: /opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.6.2/bin/chef-solo:25:in `<top (required)>'
==> default: /usr/bin/chef-solo:23:in `load'
==> default: /usr/bin/chef-solo:23:in `<main>'
==> default: Chef Client failed. 0 resources updated
==> default: [2015-02-02T15:31:52+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

**Comment by ruebot** [comment]
What do you get from these commands for comparison?

```
$ ruby -v
ruby 2.1.3p242 (2014-09-19 revision 47630) [x86_64-linux]
```

```
$ gem -v
2.2.2
```

```
$ vagrant --version
Vagrant 1.6.5
```

```
$ berks --version
3.2.0
```

```
$ vagrant plugin list
vagrant-berkshelf (2.0.1)
  - Version Constraint: = 2.0.1
vagrant-login (1.0.1, system)
vagrant-omnibus (1.4.1)
vagrant-share (1.1.2, system)
```

**Comment by daniel-dgi** [comment]
daniels-mbp:islandora_chef danny$ ruby -v
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-darwin13]

daniels-mbp:islandora_chef danny$ gem -v
2.4.5

daniels-mbp:islandora_chef danny$ vagrant --version
Vagrant 1.7.2

daniels-mbp:islandora_chef danny$ berks --version
3.2.3

daniels-mbp:islandora_chef danny$ vagrant plugin list
vagrant-berkshelf (2.0.1)
  - Version Constraint: = 2.0.1
vagrant-omnibus (1.4.1)
vagrant-share (1.1.3, system)
vagrant-vsphere (1.0.1)

**Comment by daniel-dgi** [comment]
``` 
daniels-mbp:islandora_chef danny$ ruby -v 
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-darwin13] 
```

``` 
daniels-mbp:islandora_chef danny$ gem -v 
2.4.5 
```

``` 
daniels-mbp:islandora_chef danny$ vagrant --version 
Vagrant 1.7.2 
```

``` 
daniels-mbp:islandora_chef danny$ berks --version 
3.2.3 
```

``` 
daniels-mbp:islandora_chef danny$ vagrant plugin list 
vagrant-berkshelf (2.0.1) 
    - Version Constraint: = 2.0.1 
vagrant-omnibus (1.4.1) 
vagrant-share (1.1.3, system) 
vagrant-vsphere (1.0.1) 
```

**Comment by daniel-dgi** [comment]
Also, noticed it looks like i'm still trying to download 12.04 as a basebox instead of 14.04.  Any idea how to clear that out?  I don't think it's the root issue, but i'll probably run into some weird gremlins because of it later.

**Comment by ruebot** [comment]
What shows up in your VirtualBox?

The recipe still uses 12.04. I'm running into some problems with using 14.04 in the recipe :cry: #1

**Comment by daniel-dgi** [comment]
It's still 12.04.  I can `vagrant ssh` into the box, but it's looks like just a base ubuntu install.  Nothing in /var/lib that i'd be expecting (like tomcat).  Also nothing in /opt other than a chef directory.

**Comment by ruebot** [comment]
Oh!

Try a `vagrant destroy` then a `vagrant up` and see what happens again. If you get the same thing, at least we know that :smile:

**Comment by daniel-dgi** [comment]
Turns out this is caused by the berkshelf cookbooks not being made available on the VM when it gets spun up.  By default, the islandora_chef/cookbooks folder gets mounted on the VM to get used during the install process.  This folder, of course, doesn't have access to ~/.berkshelf/cookbooks, so ka-blooey.  I worked around this by rsync'ing my ~/.berkshelf/cookbooks folder to islandora_chef/cookbooks.  I know that's hacky, and I'm sure there's a way to do all of this with configuration, but it's good enough for me to close this issue and simply log it as a known issue until someone else runs into the same error.
